hypothesis-python/src/hypothesis/_settings.py

```diff
@@ -159,22 +159,14 @@ def __getattr__(self, name):
 
     def __init__(self, parent=None, **kwargs):
         # type: (settings, **Any) -> None
-        if (
-            kwargs.get("database", not_set) is not_set
-            and kwargs.get("database_file", not_set) is not not_set
-        ):
-            if kwargs["database_file"] is None:
-                kwargs["database"] = None
-            else:
-                from hypothesis.database import ExampleDatabase
-
-                kwargs["database"] = ExampleDatabase(kwargs["database_file"])
-        if not kwargs.get("perform_health_check", True):
-            kwargs["suppress_health_check"] = HealthCheck.all()
-        if kwargs.get("max_shrinks") == 0:
-            kwargs["phases"] = tuple(
-                p for p in _validate_phases(kwargs.get("phases")) if p != Phase.shrink
-            )
+        if kwargs.get("derandomize"):
+            if kwargs.get("database") is not None:
+                # This is not a deprecation as the behaviour is not expected to
+                # change, but we still don't want this override to pass silently.
```

Comment:
I'm confused. I swear I commented on this before but GitHub seems to have eaten the comment if so.

I don't get why we want this not to be a deprecation warning - it seems like explicitly passing `database=blah, derandomize=True` will *never* do something that the user intended, so should be an error, so it seems a prime candidate for deprecation.

Reply:
I think you did... GitHub doesn't seem to handle rebasing very well 🤷‍♂️ 

- This has been deprecated since (checks) 2017, so we can just make it an error now if you prefer.
- I'd made it a warning instead because `database` defaults to a non-None value.  An error would therefore force `derandomize=True` to be paired with `database=None` every time it was used.  IMO this is poor ergonomics, so I'd prefer to delete the runtime warning and just have it documented instead.

Reply:
> I'd made it a warning instead because database defaults to a non-None value. An error would therefore force derandomize=True to be paired with database=None every time it was used.

I don't think that's true. You're checking if `database` is in kwargs, not if it's set. The thing that should be an error is `settings(database=non_none_value, derandomize=True)`, but `settings(derandomize=True)` can still just silently set database to non-None.

Reply:
:man_facepalming: yes, of course.  Will do.